test/cpp/common/alts_util_test.cc

```diff
@@ -0,0 +1,204 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <grpcpp/security/alts_context.h>
+#include <grpcpp/security/alts_util.h>
+#include <grpcpp/security/auth_context.h>
+#include <gtest/gtest.h>
+
+#include "src/core/tsi/alts/handshaker/alts_tsi_handshaker.h"
+#include "src/cpp/common/secure_auth_context.h"
+#include "src/proto/grpc/gcp/altscontext.upb.h"
+#include "test/cpp/util/string_ref_helper.h"
+
+namespace grpc {
+namespace {
+
+TEST(AltsUtilTest, EmptyAuthContext) {
+  const std::shared_ptr<AuthContext> auth_context(
+      new SecureAuthContext(nullptr));
+  // SecureAuthContext auth_context(nullptr);
+  std::unique_ptr<experimental::AltsContext> alts_context =
+      experimental::GetAltsContextFromAuthContext(auth_context);
```

Comment:
Sorry, I made a mistake in my earlier PR. The empty auth context should be created using:
    auth_context(new SecureAuthContext(ctx.get()))
instead of:
    auth_context(new SecureAuthContext(nullptr))

On the other hand, we might also want to check null auth context, in addition to the empty auth context. Another check is added for null auth context.